service/builder/exporters_builder.go

```diff
@@ -332,3 +351,22 @@ func createMetricsExporter(factoryBase component.ExporterFactoryBase,
 	// use ExporterFactoryOld.CreateMetricsExporter.
 	return factoryBase.(component.ExporterFactoryOld).CreateMetricsExporter(logger, cfg)
 }
+
+// createDataProcessor creates a data exporter based on provided factory type.
+func createDataExporter(
+	factoryBase component.ExporterFactoryBase,
+	logger *zap.Logger,
+	dataType configmodels.DataType,
+	cfg configmodels.Exporter,
+) (component.DataExporter, error) {
+	if factory, ok := factoryBase.(component.DataExporterFactory); ok {
+		creationParams := component.ExporterCreateParams{Logger: logger}
+		ctx := context.Background()
+
+		// If exporter is of the new type (can manipulate on internal data structure),
+		// use ExporterFactory.CreateDataExporter.
+		return factory.CreateExporter(ctx, creationParams, dataType, cfg)
+	}
+
+	return nil, fmt.Errorf("exporter %q does not support data type %q", factoryBase.Type(), dataType)
```

Comment:
I don't think this function really needed. For metrics and traces, it supposed to be a temporary function to support old->new style data model migration. What if we don't have `if factory, ok := factoryBase.(component.DataExporterFactory); ok {` condition? Looks like we would get a compile time error instead of this one which seems to be more useful.